src/vs/workbench/contrib/terminal/browser/terminalEditorSerializer.ts

```diff
@@ -41,7 +41,8 @@ export class TerminalInputSerializer implements IEditorSerializer {
 			titleSource: instance.titleSource,
 			cwd: '',
 			icon: instance.icon,
-			color: instance.color
+			color: instance.color,
+			resource: instance.resource.toString()
```

Comment:
Let's convert this back inside `deserialize` so that the uri <-> string conversion is contained in this file. And then remove it where we were using `string` before